### PR TITLE
Fix broken module_names in compile_dir for sub-directories

### PR DESCRIPTION
**Pull request by xforce**
Assume we have a directory structure like this:
```
test
  nested_test_1
    nested_test_2
```

Expected module names are:
```
nested_test_1
nested_test-1.nested_test-2
```

However in the actual result was:
```
nested_test_1
nested_test-1nested_test-2
```
the '.' separator missing

### CODE REVIEW

File: derive/src/compile_bytecode.rs [hunk 1]
```diff
@@ -143,7 +143,11 @@ impl CompilationSource {
             if path.is_dir() {
                 code_map.extend(self.compile_dir(
                     &path,
-                    format!("{}{}", parent, file_name),
+                    if parent.is_empty() {
+                        format!("{}{}", parent, file_name)
```
Comment (fanninpm) [inline derive/src/compile_bytecode.rs:147]:
```suggestion
                        format!("{}", file_name)
```

Comment (xforce) [reply derive/src/compile_bytecode.rs:147]: Yes. Though. Might as well just go with `file_name.to_string()` at that point. Will update the PR.

**Comment by xforce** [comment]
> oh, wow. unexpected. how can I test its wrong behavior? I added a test but the test also success for main branch. I want to add a related test for this fix.

https://github.com/xforce/rust-python-repo/blob/incorrect-module-name/src/main.rs

that code will print
```
Traceback (most recent call last):
  File "<embedded>", line 1, in <module>
ModuleNotFoundError: No module named 'nested1.nested2'
```
if you change the `from nested1.nested2` to `from nested1nested2` it works without error.
this is what this PR fixes.

**Comment by xforce** [comment]
I am at this time unfortunately not familiar enough with this codebase to provide a better test as I haven't fully explored if that particular code path is even involved in the setup that's used for tests

**Comment by youknowone** [comment]
@xforce Thank you! I added a new test based on your example

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**